powerlevel9k.zsh-theme

```diff
@@ -1730,6 +1746,20 @@ prompt_java_version() {
   fi
 }
 
+################################################################
+# Segment for Haskell Stack projects
+prompt_stack_project() {
+  local haskellstack_version=$(stack --version 2>/dev/null | grep -oE '[0-9.]+' | head -n1)
+  local stackyaml_file="stack.yaml"
+  local stackyaml_file_search=$(upsearch "${stackyaml_file}")
+
+  if [[ -n "${haskellstack_version}" ]]; then
+    if [[ "${stackyaml_file_search}" != $HOME && "${stackyaml_file_search}" != "/" ]]; then
```

Comment:
So you use `upsearch` here to show the global haskell version only if you are in a haskell project, right?

Reply:
Currently the segment displays the global Stack version, Stack being one of the most used project managers for the Haskell language. Yes, I used `upsearch` to locate the `stack.yaml` file present in Stack projects, so the segment can know if the terminal is inside a project directory or not.

I guess we could just display the string `Stack`, since the version number for Stack is not that useful, or make it display the selected compiler version for that particular project by looking into the `resolver` field of the `stack.yaml` file.

Reply:
Sounds reasonable. If that is just a project manager whose version is not interesting, then it would be better to show something else. The compiler version sounds more interesting, if it can differ between the projects.
Btw. Haskell itself is installed globally and not per project, right? May the version differ anyway (like with node and `nvm`)? Sorry for the naive question, I am absolutely clueless about Haskell. ;)